connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -96,8 +100,75 @@ public static ConnectLocation getInstance() {
      * connection is enabled and location permissions need to be granted.
      */
     public void setUpWithConnectButton(ConnectButton connectButton, LocationPermissionCallback permissionCallback) {
-        this.permissionCallback = permissionCallback;
-        connectButton.addButtonStateChangeListener(this);
+        checkInit(this);
```

Comment:
maybe we don't need to add this here? this method is called by accessing instance first, and there is already a check within that method. i.e. `ConnectLocation.getInstance().setUpWithConnectButton`